sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -584,24 +602,37 @@ class JoinSuite extends QueryTest with SharedSQLContext {
     val cartesianQueries = Seq(
       /** The following should error out since there is no explicit cross join */
       "SELECT * FROM testData inner join testData2",
-      "SELECT * FROM testData left outer join testData2",
-      "SELECT * FROM testData right outer join testData2",
-      "SELECT * FROM testData full outer join testData2",
```

Comment:
These three queries are examples for outer joins that are unable to be replaced by cross join.